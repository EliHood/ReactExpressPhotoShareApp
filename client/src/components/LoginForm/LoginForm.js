import React  from "react";
import TextField from '@material-ui/core/TextField';
import Button from '@material-ui/core/Button';

const LoginForm = ({mySubmit, myChange, username, password}) => (
    <form onSubmit={mySubmit}>
        <TextField
            label="Username"
            fullWidth
            style={{  borderRadius: 25 }}
            name="username"
            value={username}
            onChange={myChange}
            margin="normal"
            />
        <TextField
            label="Password"
            name="password"
            type="password"
            fullWidth
            style={{ borderRadius: '0px'}}
            className=""
            value={password}
            onChange={myChange}
            margin="normal"
        />
        <br></br>
        <br></br>
        <Button variant="outlined" color="primary" type="submit">
            Log In
        </Button>          
    </form>

)

export default LoginForm; 