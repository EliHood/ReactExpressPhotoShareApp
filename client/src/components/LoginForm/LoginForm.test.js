import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount } from 'enzyme';
import TextField from '@material-ui/core/TextField';
import LoginForm from './LoginForm';
import Button from '@material-ui/core/Button';
import toJson from 'enzyme-to-json';


describe('should render <LoginForm/> Component', () => {

    it('should render <LoginForm/> component', () => {
        const wrapper = shallow(<LoginForm/>)
        expect(wrapper.find('form')).toHaveLength(1);
    })

    it('should snap <LoginForm/> component', () => {
        const wrapper = shallow(<LoginForm/>)
        expect(toJson(wrapper)).toMatchSnapshot(); 
    })

    it('should render button', () => {

        const wrapper = shallow(<LoginForm/>)
        expect(wrapper.find(Button)).toHaveLength(1);
        expect(wrapper).toMatchSnapshot();

    })
})