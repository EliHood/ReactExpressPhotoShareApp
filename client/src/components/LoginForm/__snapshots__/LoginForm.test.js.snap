// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render <LoginForm/> Component should render button 1`] = `
<form>
  <WithStyles(ForwardRef(TextField))
    fullWidth={true}
    label="Username"
    margin="normal"
    name="username"
    style={
      Object {
        "borderRadius": 25,
      }
    }
  />
  <WithStyles(ForwardRef(TextField))
    className=""
    fullWidth={true}
    label="Password"
    margin="normal"
    name="password"
    style={
      Object {
        "borderRadius": "0px",
      }
    }
    type="password"
  />
  <br />
  <br />
  <WithStyles(ForwardRef(Button))
    color="primary"
    type="submit"
    variant="outlined"
  >
    Log In
  </WithStyles(ForwardRef(Button))>
</form>
`;

exports[`should render <LoginForm/> Component should snap <LoginForm/> component 1`] = `
<form>
  <WithStyles(ForwardRef(TextField))
    fullWidth={true}
    label="Username"
    margin="normal"
    name="username"
    style={
      Object {
        "borderRadius": 25,
      }
    }
  />
  <WithStyles(ForwardRef(TextField))
    className=""
    fullWidth={true}
    label="Password"
    margin="normal"
    name="password"
    style={
      Object {
        "borderRadius": "0px",
      }
    }
    type="password"
  />
  <br />
  <br />
  <WithStyles(ForwardRef(Button))
    color="primary"
    type="submit"
    variant="outlined"
  >
    Log In
  </WithStyles(ForwardRef(Button))>
</form>
`;
